### PR TITLE
first draft starlingx

### PR DESCRIPTION
**Pull request by EarthmanT**

### CODE REVIEW

File: content/working_with/official_plugins/Orchestration/starlingx.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  * `k8s_service_account_token`: The service account token, if the system has a Kubernetes Cluster.
+  * `k8s_cacert`: The Kubernetes Cluster certificate authority if the system has a Kubernetes Cluster.
+  * `k8s_admin_client_cert`: The Kubernetes Cluster client certificate if the system has a Kubernetes Cluster.
+  * `k8s_admin_client_key`: The Kubernetes Cluster client key if the system has a Kubernetes Cluster.
+  * `openstack_ip`: The system's Openstack IP if it hosts an Openstack system.
+  * `openstack_key`:  The system's Openstack key if it hosts an Openstack system.
+
+**Workflows**
+
+  * `discover_and_deploy`: Discover a System's subclouds and create deployments for all of the subclouds. One should use default values for all workflow parameters. The workflow will create a deployment group. It will then create deployments for all of the subclouds that are discovered. It will put all of the those subclouds in the group. It will then execute batch install on all of the subcloud deployments.
```
Comment (ofercloudify) [inline content/working_with/official_plugins/Orchestration/starlingx.md:104]: I think we shoudl also mention the automatic assignment of Site per discovered WRCP, and the automatic assignment of labels.

File: content/working_with/official_plugins/Orchestration/starlingx.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  * `k8s_cluster_name`: If the system has a Kubernetes cluster, the kube cluster's name.
+  * `k8s_ip`: The IP of the Kubernetes cluster, if the system has one.
+  * `k8s_service_account_token`: The service account token, if the system has a Kubernetes Cluster.
+  * `k8s_cacert`: The Kubernetes Cluster certificate authority if the system has a Kubernetes Cluster.
+  * `k8s_admin_client_cert`: The Kubernetes Cluster client certificate if the system has a Kubernetes Cluster.
+  * `k8s_admin_client_key`: The Kubernetes Cluster client key if the system has a Kubernetes Cluster.
+  * `openstack_ip`: The system's Openstack IP if it hosts an Openstack system.
+  * `openstack_key`:  The system's Openstack key if it hosts an Openstack system.
+
+**Workflows**
```
Comment (ofercloudify) [inline content/working_with/official_plugins/Orchestration/starlingx.md:102]: Is the logic regarding IPv6/4 and https relevant here at all or is it Blueprint only?

**Review by ofercloudify — CHANGES_REQUESTED** [review]

**Comment by mpeters-wrs** [comment]
Would it be more appropriate to have the Starlingx plugin under the Infrastructure Plugins rather than Orchestration?

**Comment by EarthmanT** [comment]
> Would it be more appropriate to have the Starlingx plugin under the Infrastructure Plugins rather than Orchestration?
@mpeters-wrs 
@ofercloudify What do you think?
I put it in orchestration because we are not actually modifying infra. Please let me know what you decide.

**Comment by ofercloudify** [comment]
I second @mpeters-wrs opinion here. I believe it should go under infrastructure.
While at the moment we do not modify anything within the WR systems, we certainly may in the future, and conceptually I find it closer to infra than orchestration.

**Comment by EarthmanT** [comment]
@ofercloudify @mpeters-wrs moved the page to be under infrastructure as requested.

File: content/working_with/official_plugins/Infrastructure/starlingx.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    - /plugins/starlingx/
+    - /developer/official_plugins/starlingx/
+---
+
+The StarlingX plugin enables you to discover StarlingX systems and manage resources on StarlingX systems.
+
+Start out by providing the credentials for an existing StarlingX system. The install workflow will gather information from the system, for example type and location.
+
+To discover subclouds, execute the `discover_and_deploy` workflow. This workflow performs the following tasks:
+  - Discover all subclouds that are managed by the system controller (parent).
```
Comment (ofercloudify) [inline content/working_with/official_plugins/Infrastructure/starlingx.md:17]: Missing one newline before line 17 to get the list displayed properly.

File: content/working_with/official_plugins/Infrastructure/starlingx.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+aliases:
+    - /plugins/starlingx/
+    - /developer/official_plugins/starlingx/
+---
+
+The StarlingX plugin enables you to discover StarlingX systems and manage resources on StarlingX systems.
+
+Start out by providing the credentials for an existing StarlingX system. The install workflow will gather information from the system, for example type and location.
+
+To discover subclouds, execute the `discover_and_deploy` workflow. This workflow performs the following tasks:
```
Comment (ofercloudify) [inline content/working_with/official_plugins/Infrastructure/starlingx.md:16]:
```suggestion
To discover subclouds, execute the `discover_and_deploy` workflow over a Controller deployment. This workflow performs the following tasks:
```

File: content/working_with/official_plugins/Infrastructure/starlingx.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    * `project_name`: The client project name, default: `admin`.
+    * `user_domain_name`: The client user domain name, default: `Default`.
+    * `project_domain_name`: The client project domain name, default: `Default`.
+    * `region_name`: The region name, either `RegionOne` for a system controller, or the subcloud name in the case that the system is a subcloud.
+    * `insecure`: Whether to ignore certification validation when using `https` as the protocol..
+    * `cacert`: The content of a Certificate Authority when using `https` as the protocol.
+
+  * `resource_config`: Parameters that describe the system in the StarlingX API. Currently, there is no need to provide these parameters.
+
+**Runtime Properties**:
```
Comment (ofercloudify) [inline content/working_with/official_plugins/Infrastructure/starlingx.md:66]: The flow also sets a site per Subcloud and adds it if it does not exist. not sure in which section but it is worth mentioning it here.

**Review by ofercloudify — CHANGES_REQUESTED** [review]

**Review by ofercloudify — APPROVED** [review]

**Pull request merged**